### PR TITLE
bug 1396959 Enable removal of scheduled deletes to required signoff

### PR DESCRIPTION
**Pull request by Alweezy**
Currently, there is no way to remove a scheduled delete from a role in required signoffs. This PR leverages on pre-existing API endpoints to enable this.

### CODE REVIEW

File: ui/app/templates/required_signoffs.html [hunk 1]
```diff
@@ -97,6 +97,11 @@ <h4 class="panel-title">
           <span ng-show="role['sc'] !== null">{{ role["sc"]["signoffs_required"] }} member(s) of {{ role.$key }} (currently: {{ role["signoffs_required"] }})</span>
           <span class="label label-info" ng-show="role['sc'] !== null && role['sc']['change_type'] !== 'delete'">Pending</span>
           <span class="label label-danger" ng-show="role['sc'] !== null && role['sc']['change_type'] === 'delete'">Pending Delete</span>
+          <button class="btn btn-xs btn-primary" ng-show="role['sc'] !== null && role['sc']['change_type'] === 'delete'"
+                  ng-click="undoRemoveRole(role, required_signoffs)">
+            Undo delete
```
Comment (mozbhearsum) [inline ui/app/templates/required_signoffs.html]: "Cancel delete" would be better text for this button.

File: ui/app/js/controllers/required_signoffs_controller.js [hunk 2]
```diff
@@ -43,6 +43,33 @@ function($scope, $modal, $q, ProductRequiredSignoffs, PermissionsRequiredSignoff
...
+    $scope.required_signoffs[product].channels[channel][roleName].sc.signoffs_required = 1;
+    $scope.required_signoffs[product].channels[channel][roleName].sc.change_type = "update";
+    var update_sc_id = $scope.required_signoffs[product].channels[channel][roleName].sc.sc_id;
+    var data = {
+      "product": $scope.product, 
+      "role": roleName,
+      "data_version": $scope.required_signoffs[product].channels[channel][roleName].data_version};
+    data["when"] = new Date().getTime() + 5000;
+    data["sc_data_version"] = $scope.required_signoffs[product].channels[channel][roleName].sc.sc_data_version;
+    data["signoffs_required"] = $scope.required_signoffs[product].channels[channel][roleName].sc.signoffs_required;
```
Comment (mozbhearsum) [inline ui/app/js/controllers/required_signoffs_controller.js]: I think you've done a bit more work here than is necessary. The only things that you need to include to delete a scheduled change are sc_id, data_version (which comes from sc_data_version), and csrf_token - so you should be able to remove a bunch of these lines of code.

File: ui/app/js/controllers/required_signoffs_controller.js [hunk 3]
```diff
@@ -43,6 +43,33 @@ function($scope, $modal, $q, ProductRequiredSignoffs, PermissionsRequiredSignoff
     }
   }, true);
 
+  $scope.undoRemoveRole = function(role, rs, csrf_token) {
```
Comment (mozbhearsum) [inline ui/app/js/controllers/required_signoffs_controller.js]: Does csrf_token actually do anything here?

File: ui/app/js/controllers/required_signoffs_controller.js [hunk 4]
```diff
@@ -43,6 +43,33 @@ function($scope, $modal, $q, ProductRequiredSignoffs, PermissionsRequiredSignoff
...
+    var roleName = role.$key;
```
Comment (mozbhearsum) [inline ui/app/js/controllers/required_signoffs_controller.js]: Although $key is accessible here, it's not safe to refer to it. You should pass it in as another argument instead, if you truly need it.

File: ui/app/js/controllers/required_signoffs_controller.js [hunk 5]
```diff
@@ -43,6 +43,27 @@ function($scope, $modal, $q, ProductRequiredSignoffs, PermissionsRequiredSignoff
     }
   }, true);
 
+  $scope.undoRemoveRole = function(roleName, rs) {
+    var product = Object.keys(rs)[0];
+    var channel = Object.keys(rs[product].channels)[0];
+
+    $scope.required_signoffs[product].channels[channel][roleName].sc.signoffs_required = 1;
```
Comment (mozbhearsum) [inline ui/app/js/controllers/required_signoffs_controller.js]: Does this serve a purpose? You're setting sc to null if the http request succeeds a few lines down.

Comment (Alweezy) [reply ui/app/js/controllers/required_signoffs_controller.js]: @mozbhearsum The line was serving no purpose for sure, I have since removed it. Thanks.

**Pull request merged**